tests/components/fan/common.py

```diff
@@ -146,3 +149,27 @@ async def async_set_direction(
 
     await hass.services.async_call(DOMAIN, SERVICE_SET_DIRECTION, data, blocking=True)
     await hass.async_block_till_done()
+
+
+class MockFan(MockEntity, FanEntity):
+    """Mock Fan class."""
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return preset mode."""
+        return self._handle("preset_mode")
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return preset mode."""
+        return self._handle("preset_modes")
+
+    @property
+    def supported_features(self):
```

Comment:
Noticed now that the mock entity already implements supported features.

Reply:
Fixed in https://github.com/home-assistant/core/pull/114425